### ISSUE TITLE
#FeatureRequest adding support for importing from MKD_ARM Keil project files

### ISSUE THREAD

**Issue by ali80** [damogranlabs/VS-Code-STM32-IDE#2]
I believe this feature has the following benefits:
- you are not limited to projects created by cube
- you are not limited to STM32 MCUs
- you can add files, or reconfigure project mid development using keil.
- you can import your existing keil projects
- you can use Keil for things like debugging which may not be completely ready on VSCode yet.
- suddenly changing your IDE is risky and costly for a lot of developers, this feature can remove this risk 
and developers can decide which is best for them and can go back any time they want this can accelerate the adaptation of VSCode a lot which leads to faster development in its embedded modules as well.

**Comment by schperplata** [comment]
Well, Keil project importer was the idea from the very beginning, since amost all Damogran Labs projects were based on Keil. I often wish to just fix some minor bug and I can't recompile it, because I don't have all dependencies or it is a version mismatch or I hit code size limit and so on... So yes, work in progress, but no specific ETA.

But I only thought about one way, this is: Keil -> VS Code, never back to Keil. In fact, the whole idea of this project was to move away from restricted and paid stuff. As you said, VS Code debug capabilities so far are quite limited, so I am investigating another option for advanced debug capabilities. 

Another problem is reading/parsing/writing/appending data of the Makefile. The syntax of Makefiles is known and common, but currently parser and modifier script is tightly adapted to STM32 Cube generated Makefile. If someone would wish to implement support for other CPUs, Makefile structure and concept is actually the only problem that would take time. 

Bottom line:
* Keil project importer script will eventually be available.
* Advanced debug guide will probably also be available soon.
* Currently I mainly use STM32, so I don't intend to implement support for any other CPUs.

**Comment by ali80** [comment]
> But I only thought about one way, this is: Keil -> VS Code, never back to Keil. 

I meant the same.

>Another problem is reading/parsing/writing/appending data of the Makefile.

I'm wild guessing that this should be easier in Keil project since it has a standard format.
anyway, this is great news, hope the bugs get resolved soon :)

**Comment by schperplata** [comment]
[Keil import script available. ](https://github.com/damogranlabs/VS-Code-STM32-IDE/blob/master/README_KEIL_IMPORTER.md)
